src/goto-programs/remove_exceptions.cpp

```diff
@@ -559,22 +514,26 @@ void remove_exceptionst::instrument_exceptions(
     else if(instr_it->type==THROW)
     {
       instrument_throw(
-        function_id, goto_program, instr_it, stack_catch, locals);
+        goto_program, instr_it, stack_catch, locals);
     }
     else if(instr_it->type==FUNCTION_CALL)
     {
       instrument_function_call(
-        function_id, goto_program, instr_it, stack_catch, locals);
+        goto_program, instr_it, stack_catch, locals);
     }
   }
 }
 
 void remove_exceptionst::operator()(goto_functionst &goto_functions)
 {
   Forall_goto_functions(it, goto_functions)
-    add_exceptional_returns(it->first, it->second.body);
-  Forall_goto_functions(it, goto_functions)
-    instrument_exceptions(it->first, it->second.body);
+    instrument_exceptions(it->second.body);
+  if(symbol_table.has_symbol(INFLIGHT_EXCEPTION_VARIABLE_NAME))
+  {
+    auto findit = goto_functions.function_map.find(INITIALIZE_FUNCTION);
+    if(findit != goto_functions.function_map.end())
+      initialize_inflight_exception_global(findit->second.body);
+  }
```

Comment:
See above: I believe this symbol should be added deterministically by the language front-end.

Reply:
Hmmm. The Java frontend produces "high-level" exceptions, which use throw, catch etc and don't have an explicit in-flight exception variable. This pass converts (lowers ;)) the high-level representation to explicit ops on an inflight-exception global, so I maintain it ought to be introduced here.

Reply:
While (as you proved) you _can_ introduce it here, the code would be a lot simpler if you didn't. Add it in `java_bytecode_internal_additions.cpp` and enjoy.

Reply:
Alright, but I'm not a fan since that means the frontend will introduce something that may be unused, depending on whether or not this pass is scheduled for later in the pipeline. Oh well, small cost.